### PR TITLE
Move most task callback support into common code.

### PR DESCRIPTION
**Pull request by gbtitus**
The bulk of task callback support, minus common type definitions and
such, had been implemented separately in tasks=fifo and =qthreads.  Here
we move most of it into common code.  The functions that users call to
install and uninstall callbacks, and the one the tasking layers call to
actually do the callbacks, are in chpl-tasks-callbacks.c.  Definitions
for the latter and wrappers that can be inlined to reduce overhead are
in chpl-tasks-callbacks-internal.h.  All of the callback-related code
except the calls to actually do the callbacks is removed from the
qthreads and fifo tasking layers.

### CODE REVIEW

**Comment by gbtitus** [comment]
Hey @ronawho , got time for this one?  (It's just structural, not functional.)

File: runtime/include/chpl-tasks-callbacks-internal.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+
+//
+// Don't refer to this directly.  It's only here in public to support
+// reducing overhead by inlining the test for >0.
+//
+int chpl_task_callback_counts[chpl_task_cb_num_event_kinds];
+
+
+void chpl_task_do_callbacks_really(chpl_task_cb_event_kind_t,
```
Comment (ronawho) [inline runtime/include/chpl-tasks-callbacks-internal.h]: Think I'd prefer _internal to _really, but not strongly

**Comment by ronawho** [comment]
:+1: :+1: :+1:

File: runtime/include/chpl-tasks-callbacks-internal.h (hunk 1 above)
Comment (gbtitus) [inline runtime/include/chpl-tasks-callbacks-internal.h]: Yeah, that's better. I'll make that change before merging.

**Pull request merged**